### ISSUE TITLE
Extend documentation with CMake options

### ISSUE THREAD

**Issue by ntamas** [igraph/igraph#1812]
**What is the feature or improvement you would like to see?**

The installation part in the documentation should be extended with a listing of the most common CMake options.

**Use cases for the feature**

E.g., turning off `IGRAPH_WARNINGS_AS_ERRORS` is particularly useful for downstream projects that compile igraph as part of their toolchain if the toolchain itself throws warnings that we did not anticipate.

**References**

N/A

**Comment by szhorvat** [comment]
I was thinking about fixing this yesterday, and one thought I had was to not put it directly into the installation instructions. Instead, create a Troubleshooting (or Possible Issues) section, and mention it there, along with other potential problems.

We don't want most people (e.g. distro maintainers) to set this to OFF by default. Ideally, if a warning is triggered, people would come and report it to us.  The risk of documenting it along with more common options is that people will just set it to OFF from the get-go.

Thoughts?

**Comment by vtraag** [comment]
I agree that separating it from the main instructions is a good idea. We could also rubric it under something like "Advanced Configuration" or "Advanced Settings" or the like? A "Troubleshooting" or "Possible Issues" section also sounds good to me, but it might be nice to simply have a list of possible configuration options, instead of necessarily looking at the troubleshooting section to look for configuration options.

**Comment by ntamas** [comment]
I'd probably go for an "Advanced Configuration" settings chapter, with a big disclaimer at the beginning ("Abandon all expectations, ye who enter here"). Not all CMake options are related to troubleshooting; e.g., the possibility to turn off GraphML support to spare a dependency on `libxml2` is not "troubleshooting", and neither is the option to configure igraph as 32-bit on a 64-bit platform.

**Comment by iosonofabio** [comment]
Sounds like a great idea. If you do want to use that disclaimer, just beware the original Italian says "Abandon all hope..." 😉

**Comment by ntamas** [comment]
I know but I felt that it wouldn't be appropriate here after all :)

**Comment by szhorvat** [comment]
I won't fix this for 0.9.5. Almost all options are already documented:

https://igraph.org/c/doc/igraph-Installation.html#igraph-Installation-notable-configuration-options

A GUI (either ccmake or cmake-gui) will display relevant options, including `IGRAPH_WARNINGS_AS_ERRORS`, with a short description. Thus, the option is discoverable.

We can restructure the docs with an Advanced Configuration section for 0.10. Right now this would be the only "advanced" option.

**Comment by szhorvat** [comment]
@ntamas `BUILD_TESTING`: The tests are not built unless one is explicitly asking for it (building the `build_tests` target, or something that depends on it such as `test` / `check`). Why then do we document it?  

I think we should remove it as the current text is confusing. It suggests that the tests are built by default, which is not true.
